boards: nordic: nrf54h20dk: Add cpuapp TCM memory section

nordic-krch opened the pull request:
Add memory section which is placed in cpuapp TCM (Tightly Coupled Memory) RAM0 which is fast, non-cacheable.

~~Contains #100564 which fixes failing CI.~~

nordic-krch committed:
boards: nordic: nrf54h20dk: Add cpuapp TCM memory section

Add memory section which is placed in cpuapp TCM (Tightly Coupled
Memory) RAM0 which is fast, non-cacheable.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

tomi-font reviewed (changes requested):
nacking to make sure this doesn't get merged before we get a proper fix for https://github.com/zephyrproject-rtos/zephyr/issues/100781

nordic-krch commented:
@tomi-font can you remove the NACK, I've removed the commit and rebased.
@bjarki-andreasen can you re-approve? No changes in the the commit only rebase and CI fix commit removal.

nordic-krch commented:
@tomi-font can you remove the nack? commit is removed.

tomi-font commented:
Fail? :upside_down_face:
